### PR TITLE
Feature 328 release dropdown menu

### LINKED ISSUE #328: Enhance the Release Notes by adding dropdown menus
Please use [Sphinx Design for Dropdown menus](https://sphinx-design.readthedocs.io/en/latest/dropdowns.html) . This will allow for searches of material hidden within dropdown menus.

Changes will need to be made to the below files:
1.  config.py 
    add 'sphinx_design' to the "extensions =" section.  (note the underscore.) 
2.  docs/requirements.txt file. 
    add 'sphinx-design==0.3.0' with a dash 
3. METplotpy/.github/workflows/documentation.yml 
    add a line after this example [line 28](https://github.com/dtcenter/METviewer/blob/505660418afbc1debb301c5f0b4bcd28823b3896/.github/workflows/documentation.yml#L28) . Make sure it is correctly indented.
   python -m pip install -r docs/requirements.txt

It should look like this:

          python -m pip install --upgrade python-dateutil requests sphinx \
          sphinx-gallery matplotlib Pillow sphinx_rtd_theme
          python -m pip install -r docs/requirements.txt

4.  Change  python-version: '3.10' 
     back to  python-version: '3.8' in the METcalcpy/.github/workflows/documentation.yml file

The dropdown menus won't work unless this is added.

Panel drop downs would be added for the subcategories and sub-subcategories. For example,
-Repository and build
---Installation
---Static Code Analysis
---Testing
---Continuous Integration
-Documentation
-Library code
---Bugfixes
---Python embedding enhancements
---Miscellaneous
---NetCDF Library
---Statistics computations
etc.

## Describe the Task ##
*Provide a description of the task here.*

### Time Estimate ###
*less than a day*

### Sub-Issues ###
None

### Relevant Deadlines ###
*NONE.*

### Funding Source ###
*Split between accounts 2702691 and 2792542.*

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [x] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Task Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issues
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by lisagoodrich**
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
I've added dropdown menus to the release note section. The below files have been changed so the dropdown menus will work:
config.py
requirements.txt file.
METplotpy/.github/workflows/documentation.yml

I have tested all of the dropdown menus. 

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Please open a couple of the dropdown menus.  Let me know if there are any problems or inconsistencies.  All should be running smoothly.  Here is the link:
https://metplotpy.readthedocs.io/en/feature_328_release_dropdown_menu/Users_Guide/release-notes.html

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[No errors]**

- [x] Do these changes include sufficient testing updates? **[No]**

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Please complete this pull request review by **[4/4/23]**.</br> Date is negotiable.

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Review the source issue metadata (required labels, projects, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select **Development** issue with the original issue number.
- [x] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [x] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -15,3 +15,4 @@ scikit-image==0.19.3
 scipy==1.9.3  
 xarray==2023.1.0
 eofs==1.4.0
+sphinx-design==0.3.0
```
Comment (jprestop) [inline requirements.txt]:
```suggestion
```

Comment (jprestop) [reply requirements.txt]: Please remove "sphinx-design==0.3.0" from the requirements.txt file.  It only needs to be in the docs/requirements.txt as it is only a requirement to build the documentation and not a requirement for using METplotpy.

**Comment by lisagoodrich** [comment]
@jprestop Where is the requirements.rst file?  Do I need to create this?

**Comment by jprestop** [comment]
@lisagoodrich 

> @jprestop Where is the requirements.rst file? Do I need to create this?

It looks like no further action is needed here.  You committed my suggested change for requirements.txt and I don't see sphinx-design in requirements.txt any longer.  But, for reference, requirements.txt is in the top level METplotpy directory.

**Review by jprestop — APPROVED** [review]
I approve this request.

**Comment by lisagoodrich** [comment]
Hello @fisherhucar & @bikegeek I don't have permissions to squash and merge these changes. I'm done with the branch and the issue can be closed. Thanks!

**Comment by jprestop** [comment]
@bikegeek, upon the merge to develop, I see there was a "Python vulnerability check" failure.   Here is the GitHub Action failure link:
https://github.com/dtcenter/METplotpy/actions/runs/4533998715

I am not sure what to do to resolve this problem.  Thanks for any help you can give.

**Comment by lisagoodrich** [comment]
@jprestop, @bikegeek has updated the opencv-python versions to 4.7.0.72 and that addressed the failed vulnerability checks.   Do you want to try merging again?

**Comment by jprestop** [comment]
@lisagoodrich Merging can only be done once and it was already done.  We can see from the GitHub Actions page:
https://github.com/dtcenter/METplotpy/actions
that the vulnerability checks are now passing for "develop" which this feature branch was merged into, so no further action needs to be taken.

**Pull request merged**